neuralpredictors/measures/np_functions.py

```diff
@@ -12,10 +15,152 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
+        _, r, n = repeated_outputs.shape
+        oracles = (repeated_outputs.mean(axis=1, keepdims=True) - repeated_outputs / r) * r / (r - 1)
+        if np.any(np.isnan(oracles)):
+            logger.warning(
+                "{}% NaNs when calculating the oracle. NaNs will be set to Zero.".format(np.isnan(oracles).mean() * 100)
+            )
+        oracles[np.isnan(oracles)] = 0
+        return corr(oracles.reshape(-1, n), repeated_outputs.reshape(-1, n), axis=0)
+    else:
+        oracles = []
+        for outputs in repeated_outputs:
+            r, n = outputs.shape
+            # compute the mean over repeats, for each neuron
+            mu = outputs.mean(axis=0, keepdims=True)
+            # compute oracle predictor
+            oracle = (mu - outputs / r) * r / (r - 1)
+
+            if np.any(np.isnan(oracle)):
+                logger.warning(
+                    "{}% NaNs when calculating the oracle. NaNs will be set to Zero.".format(
+                        np.isnan(oracle).mean() * 100
+                    )
+                )
+                oracle[np.isnan(oracle)] = 0
+
+            oracles.append(oracle)
+        return corr(np.vstack(repeated_outputs), np.vstack(oracles), axis=0)
+
+
+def explainable_var(repeated_outputs, eps=1e-9):
+    """
+    Compute the explainable variance per neuron.
+
+    Args:
+        repeated_outputs (array): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    img_var = []
+    total_var = np.var(np.vstack(repeated_outputs), axis=0, ddof=1)
+    for out in repeated_outputs:
+        img_var.append(np.var(out, axis=0, ddof=1))
+    img_var = np.vstack(img_var)
+    noise_var = np.mean(img_var, axis=0)
+    explainable_var = (total_var - noise_var) / (total_var + eps)
+    return explainable_var
+
+
+def fev(targets, predictions, return_exp_var=False):
+    """
+    Compute the fraction of explainable variance explained per neuron
+
+    Args:
+        targets (list): Neuronal responses (ground truth) to image repeats. Dimensions:
+            [num_images] np.array(num_repeats, num_neurons)
+        outputs (list): Model predictions to the repeated images, with an identical shape as the targets
+        return_exp_var (bool): returns the fraction of explainable variance per neuron if set to True
+    Returns:
+        FEVe (np.array): the fraction of explainable variance explained per neuron
+        --- optional: FEV (np.array): the fraction
+    """
+
+    img_var = []
+    pred_var = []
+    for i, _ in enumerate(targets):
```

Comment:
rather than doing `targets[i]` consider rewriting as follows:

```python
from itertools import count
for i, target, prediction in zip(count(), targets, predictions):
    pred_var.append((target - prediction) **2)
    img_var.append(np.var(targets, axis=0, ddof=1))
```